test/unit/src/lib/components/LivestreamContainer/index.js

```diff
@@ -0,0 +1,105 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {assert} = require('../../assert');
+require('../../../../../../src/lib/components/LivestreamContainer/index');
+
+describe('LivestreamContainer', function () {
+  let container;
+
+  before(async function () {
+    await customElements.whenDefined('web-livestream-container');
+
+    container = document.createElement('web-livestream-container');
+    document.body.append(container);
+    container.onStateChanged({activeEventDay: null, isSignedIn: false});
+  });
+
+  after(() => {
+    container.remove();
+  });
+
+  it('should render an iframe with a videoId', async function () {
+    await container.updateComplete;
+    assert(
+      container.renderRoot.querySelector('iframe') === null,
+      'no iframe should be rendered without videoId',
+    );
+
+    container.onStateChanged({
+      activeEventDay: {
+        videoId: 'HtTyRajRuyY',
+        isChatActive: false,
+      },
+      isSignedIn: false,
+    });
+    await container.updateComplete;
+
+    const ytIframe = container.renderRoot.querySelector('iframe');
+    assert(ytIframe !== null, 'iframe should render');
+
+    const iframes = container.renderRoot.querySelectorAll('iframe');
+    assert(iframes.length === 1, 'should be single iframe');
+
+    container.onStateChanged({
+      activeEventDay: {
+        videoId: 'HtTyRajRuyY',
+        isChatActive: true,
+      },
+      isSignedIn: false,
+    });
+    await container.updateComplete;
+
+    const iframesWithChat = container.renderRoot.querySelectorAll('iframe');
+    assert(iframesWithChat.length === 2, 'should be two iframes');
+    assert(iframesWithChat[0] === ytIframe, 'YouTube iframe should not change');
+  });
+
+  it('should reload on signed-in state change', async function () {
+    container.onStateChanged({
+      activeEventDay: {
+        videoId: 'HtTyRajRuyY',
+        isChatActive: false,
+      },
+      isSignedIn: true,
+    });
+    await container.updateComplete;
+
+    const ytIframe = container.renderRoot.querySelector('iframe');
+    assert(ytIframe !== null, 'iframe should render');
+
+    // nb. This works because our tests run with sandbox mode off. We can happily talk between
+    // cross-origin frames and listen to their events.
+    const p = new Promise((resolve) => {
+      ytIframe.contentWindow.addEventListener('unload', resolve, {once: true});
+    });
+
+    container.onStateChanged({
+      activeEventDay: {
+        videoId: 'HtTyRajRuyY',
+        isChatActive: false,
+      },
+      isSignedIn: false,
+    });
+    await container.updateComplete;
+
+    const timeout = new Promise((r) => setTimeout(r, 2000)).then(() => {
+      throw new Error("unload event didn't arrive for forced reload");
+    });
+    await Promise.race([p, timeout]);
```

Comment:
instead of racing is it possible to await `p` and assert that it resolves? Just in case, for whatever reason, the timeout ends up being flaky.

I think mocha has a built in 5s timeout so if the assert doesn't get triggered within that time then it would produce a similar behavior to the race you have now.